### PR TITLE
Remove cudf::strings::repeat_strings_output_sizes and optional parameter from cudf::strings::repeat_strings

### LINKED ISSUE #12542: Deprecate cudf::strings::repeat_strings_output_sizes in favor of cudf::repeat_strings throwing size limit error
The `cudf::strings::repeat_strings_output_sizes` was created so the output sizes of the `cudf::strings::repeat_strings` API could be pre-computed and checked for possible overflow. The `cudf::strings::repeat_strings` can now throw an exception if the output size will exceed the column limit size so this API is no longer needed.

The following API can be removed: https://docs.rapids.ai/api/libcudf/stable/group__strings__copy.html#gad137eea9f5189b3e8c6384c6ab8def40
The optional parameter for this API can be removed: https://docs.rapids.ai/api/libcudf/stable/group__strings__copy.html#ga89b08a3e8f941f7760087cb9e5bf9758

@revans2

### PR DESCRIPTION
**Pull request by davidwendt**
## Description
Removes `cudf::strings::repeat_strings_output_sizes` and the optional sizes parameter from `cudf::strings::repeat_strings`.
This function (and corresponding optional parameter) is no longer needed now that the internal utilities will throw an error if the column output size exceeds the maximum.
Closes #12542

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/src/strings/repeat_strings.cu [hunk 1]
```diff
@@ -189,146 +189,63 @@ struct compute_size_and_repeat_separately_fn {
   // If d_chars != nullptr: only repeat strings.
   char* d_chars{nullptr};
 
-  __device__ int64_t operator()(size_type const idx) const noexcept
+  __device__ void operator()(size_type const idx) const noexcept
```
Comment (karthikeyann) [inline cpp/src/strings/repeat_strings.cu:192]: is this a `const` function still?

Comment (davidwendt) [reply cpp/src/strings/repeat_strings.cu:192]: Yes, the member variables are not changed by this function.

Comment (karthikeyann) [reply cpp/src/strings/repeat_strings.cu:192]: should writes to `d_offsets[idx]` and `output_ptr` be not considered for this?

Comment (karthikeyann) [reply cpp/src/strings/repeat_strings.cu:192]: https://stackoverflow.com/questions/7390437/c-const-member-functions-are-modifying-member-variables

got it.

Comment (davidwendt) [reply cpp/src/strings/repeat_strings.cu:192]: That's correct. The function does not change the actual `d_offsets` or `d_chars` variables. Changing data/memory they point to does not count.

**Review by karthikeyann — APPROVED** [review]

File: cpp/include/cudf/strings/repeat_strings.hpp [hunk 2]
```diff
@@ -32,15 +32,15 @@ namespace strings {
  */
 
 /**
- * @brief Repeat the given string scalar by a given number of times.
+ * @brief Repeat the given string scalar by a given number of times
```
Comment (bdice) [inline cpp/include/cudf/strings/repeat_strings.hpp:35]: “Repeat by” sounds odd to me.
```suggestion
 * @brief Repeat the given string scalar a given number of times
```

File: cpp/include/cudf/strings/repeat_strings.hpp [hunk 3]
```diff
@@ -50,33 +50,30 @@ namespace strings {
...
+ * @return New string scalar in which the input string is repeated
  */
 std::unique_ptr<string_scalar> repeat_string(
   string_scalar const& input,
   size_type repeat_times,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
 /**
- * @brief Repeat each string in the given strings column by a given number of times.
+ * @brief Repeat each string in the given strings column by a given number of times
```
Comment (bdice) [inline cpp/include/cudf/strings/repeat_strings.hpp:67]:
```suggestion
 * @brief Repeat each string in the given strings column a given number of times
```

**Review by bdice — APPROVED** [review]
Thanks! Good cleanup.

**Pull request merged**